lib/permutations.rb

```diff
@@ -1,4 +1,25 @@
 
 def permutations?(string1, string2)
-  raise NotImplementedError, "permutations? not implemented"
-end
\ No newline at end of file
+  
+  if string1.empty? || string2.empty?
+    return true
+  elsif string1.length != string2.length
+    return false
+  end
+
+
+  i = string1.chars
+  j = string2.chars
```

Comment:
instead of getting whole new arrays of characters you can use `.each_char` to loop through the strings letter-by-letter.